### ISSUE TITLE
Add more automatic labelled examples for the stepstoreproduce model

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#426]
We should add the following to the training set of stepstoreproduce:
1) Bugs that had 'stepswanted' keyword (rollbacked to the time before the keyword was removed)
2) Bugs that had 'Has STR' set to no (rollbacked to the time before 'Has STR' turned to yes)

This is not really easy to implement. The bugs that had 'Has STR=no' and now have 'Has STR=yes' are already in the training set as examples of bugs *with* steps to reproduce.
So, we'll need to consider some bugs twice, once rollbacked and once with the full history.

**Comment by chidauri** [comment]
@marco-c 
__________________________________________________________________________________Before changes__________________________________________________________________________________

70 bugs have no steps to reproduce
2356 bugs have steps to reproduce
X: (2426, 61217), y: (2426,)
Cross Validation scores:
Accuracy: f0.7150714840551717 (+/- 0.05481197137931066)
Precision: f0.989552280294159 (+/- 0.009526694711771396)
Recall: f0.7138799678843839 (+/- 0.05667355088157277)
X_train: (130, 61217), y_train: (130,)
X_test: (243, 61217), y_test: (243,)
Top 26 features:
+-----------+----------------------+---------------------+------------------------+-----------------------+----------------+------------------+--------------------+
|   classes |   comments__expected |   comments__provide |   comments__attachment |   comments__reproduce |   comments__as |   comments__some |   comments__should |
+===========+======================+=====================+========================+=======================+================+==================+====================+
|         0 |            0.0757799 |          -0.0549655 |             -0.0529733 |            -0.0525551 |      0.0448003 |        -0.042769 |          0.0388465 |
+-----------+----------------------+---------------------+------------------------+-----------------------+----------------+------------------+--------------------+

+-------------------+-----------------------+---------------------+----------------+----------------------------------+----------------+-----------------+--------------------+
|   comments__right |   comments__duplicate |   comments____url__ |   comments__if |   data__has_regression_range=yes |   comments__is |   comments__can |   comments__result |
+===================+=======================+=====================+================+==================================+================+=================+====================+
|         0.0368999 |             0.0331057 |          -0.0330953 |     -0.0303417 |                        0.0281634 |     -0.0244752 |      -0.0238259 |          0.0227891 |
+-------------------+-----------------------+---------------------+----------------+----------------------------------+----------------+-----------------+--------------------+

+------------------+-----------------+----------------+----------------+-------------------+----------------+----------------+-----------------+
|   comments__help |   comments__fix |   comments__it |   comments__in |   comments__build |   comments__bp |   comments__to |   comments__are |
+==================+=================+================+================+===================+================+================+=================+
|       -0.0226297 |      -0.0224083 |     -0.0199891 |      0.0198101 |         0.0191122 |     -0.0170332 |      0.0151535 |      -0.0123306 |
+------------------+-----------------+----------------+----------------+-------------------+----------------+----------------+-----------------+

+------------------+-----------------+----------------+
|   comments__that |   comments__but |   comments__us |
+==================+=================+================+
|       -0.0122262 |       0.0121193 |     -0.0118019 |
+------------------+-----------------+----------------+

Test Set scores:
No confidence threshold - 243 classified
[[179  59]
 [  3   2]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.98      0.75      0.40      0.85      0.55      0.31       238
          0       0.03      0.40      0.75      0.06      0.55      0.29         5

avg / total       0.96      0.74      0.41      0.84      0.55      0.31       243


Confidence threshold > 0.6 - 212 classified
[[166  41]
 [  3   2]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.98      0.80      0.40      0.88      0.57      0.33       207
          0       0.05      0.40      0.80      0.08      0.57      0.31         5

avg / total       0.96      0.79      0.41      0.86      0.57      0.33       212


Confidence threshold > 0.7 - 179 classified
[[141  33]
 [  3   2]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.98      0.81      0.40      0.89      0.57      0.34       174
          0       0.06      0.40      0.81      0.10      0.57      0.31         5

avg / total       0.95      0.80      0.41      0.86      0.57      0.34       179


Confidence threshold > 0.8 - 135 classified
[[116  17]
 [  1   1]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.99      0.87      0.50      0.93      0.66      0.45       133
          0       0.06      0.50      0.87      0.10      0.66      0.42         2

avg / total       0.98      0.87      0.51      0.92      0.66      0.45       135


Confidence threshold > 0.9 - 82 classified
[[72  9]
 [ 0  1]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.89      1.00      0.94      0.94      0.88        81
          0       0.10      1.00      0.89      0.18      0.94      0.90         1

_______________________________________________________________________________________after changes________________________________________________________________________________--

70 bugs have no steps to reproduce
2356 bugs have steps to reproduce
X: (2438, 61217), y: (2438,)
Cross Validation scores:
Accuracy: f0.7269936863564972 (+/- 0.1172795443670644)
Precision: f0.9841704306930333 (+/- 0.009017260788516033)
Recall: f0.7292452830188679 (+/- 0.12477819409499849)
X_train: (148, 61217), y_train: (148,)
X_test: (244, 61217), y_test: (244,)
Top 18 features:
+-----------+--------------------+---------------------+----------------+------------------------+----------------------------------+---------------------+-----------------+
|   classes |   comments__should |   comments__provide |   comments__as |   comments__attachment |   data__keywords-regression=True |   comments____url__ |   comments__are |
+===========+====================+=====================+================+========================+==================================+=====================+=================+
|         0 |          0.0966837 |          -0.0483236 |      0.0425127 |             -0.0423763 |                        -0.039338 |          -0.0283389 |       0.0275511 |
+-----------+--------------------+---------------------+----------------+------------------------+----------------------------------+---------------------+-----------------+

+----------------+----------------+----------------------+-----------------+-----------------+-----------------+----------------+------------------+
|   comments__to |   comments__in |   comments__expected |   comments__bug |   comments__fix |   comments__see |   comments__do |   comments__from |
+================+================+======================+=================+=================+=================+================+==================+
|      0.0275017 |      0.0273187 |            0.0268631 |       0.0258993 |      -0.0249801 |       0.0200248 |     -0.0191712 |        0.0190359 |
+----------------+----------------+----------------------+-----------------+-----------------+-----------------+----------------+------------------+

+------------------+-------------------+-----------------+
|   comments__with |   comments__which |   comments__str |
+==================+===================+=================+
|        0.0177384 |         0.0164346 |      -0.0152949 |
+------------------+-------------------+-----------------+

Test Set scores:
No confidence threshold - 244 classified
[[170  66]
 [  0   8]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.72      1.00      0.84      0.85      0.70       236
          0       0.11      1.00      0.72      0.20      0.85      0.74         8

avg / total       0.97      0.73      0.99      0.82      0.85      0.70       244


Confidence threshold > 0.6 - 208 classified
[[151  50]
 [  0   7]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.75      1.00      0.86      0.87      0.73       201
          0       0.12      1.00      0.75      0.22      0.87      0.77         7

avg / total       0.97      0.76      0.99      0.84      0.87      0.73       208


Confidence threshold > 0.7 - 170 classified
[[128  36]
 [  0   6]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.78      1.00      0.88      0.88      0.76       164
          0       0.14      1.00      0.78      0.25      0.88      0.80         6

avg / total       0.97      0.79      0.99      0.85      0.88      0.76       170


Confidence threshold > 0.8 - 129 classified
[[97 26]
 [ 0  6]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.79      1.00      0.88      0.89      0.77       123
          0       0.19      1.00      0.79      0.32      0.89      0.81         6

avg / total       0.96      0.80      0.99      0.86      0.89      0.77       129


Confidence threshold > 0.9 - 68 classified
[[51 14]
 [ 0  3]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.78      1.00      0.88      0.89      0.77        65
          0       0.18      1.00      0.78      0.30      0.89      0.80         3

avg / total       0.96      0.79      0.99      0.85      0.89      0.77        68



avg / total       0.99      0.89      1.00      0.93      0.94      0.88        82

**Comment by chidauri** [comment]
https://github.com/mozilla/bugbug/blob/d102ccfa54cbb5de660c0d1a9d7b4171bb9ac392/bugbug/models/stepstoreproduce.py#L77-L80
These aren't contributing anything to labels when i tested, so didn't implement Point 1 for now . These results are for Point 2 above.

**Comment by marco-c** [comment]
Thanks, so it looks like in general the number of bugs for both point 1 and point 2 is small (I see your dataset went from 2426 to 2438 bugs).
Anyway, this is worth implementing as for other models it could be important.
